### ISSUE TITLE
Implement proper way for daemons to exit on certain signals

### ISSUE THREAD

**Issue by omoerbeek** [PowerDNS/pdns#8347]
Calling exit(3) and other non-async-signal-safe code in a signal handler is not ok. Proper way is to set a (atomic) flag and exit in main loop. This is a reminder to implement it like this.

**Comment by pieterlexis** [comment]
@omoerbeek do we actually care about this?

**Comment by omoerbeek** [comment]
I do! rec already does it like that, I would have to check auth and dnsdist.

**Comment by omoerbeek** [comment]
> @omoerbeek do we actually care about this?

Yes, it makes things safer and allows for better (valgrind) testing.

**Comment by rgacogne** [comment]
dnsdist now handles that properly as well.
